src/lib/Microsoft.Health.Fhir.Ingest/Service/FhirImportService.cs

```diff
@@ -31,5 +31,7 @@ protected static (string Identifer, string System) GenerateObservationId(IObserv
         }
 
         public abstract Task ProcessAsync(ILookupTemplate<IFhirTemplate> config, IMeasurementGroup data, Func<Exception, IMeasurementGroup, Task<bool>> errorConsumer = null);
+
+        public abstract Task<string> CheckHealth();

```

Comment:
Why part of the FhirImportService?  Would ideally like this to be it's own service

Reply:
How will this get called?  Ideally it would be another service that would be wrapped in a timer based function.  https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-timer?tabs=csharp

Reply:
Yes the plan would be to wrap this in a timer based function from a non OSS repo. The reason why I put it in FhirImportService is that it would be nice to just pass in MeasurementFhirImportService measurementImportService to our timed based Health Check function and perform a health check that would mimic the same behavior/config as the real MeasurementCollectionToFhir function. 

I had originally started writing it as a separate service but found it involved copying nearly identical code in many places. I don't have a preference though.

Reply:
I think a separate service would prefer since the intended function is different.  Having a base IFhirService that takes the client that both inherit from makes sense though to reduce duplication.